cc/crypto/client_encryptor.cc

```diff
@@ -17,7 +17,73 @@
 #include "cc/crypto/client_encryptor.h"
 
 #include "absl/status/statusor.h"
+#include "cc/crypto/common.h"
 #include "cc/crypto/hpke/sender_context.h"
 #include "oak_crypto/proto/v1/crypto.pb.h"
 
-namespace oak::crypto {}  // namespace oak::crypto
+namespace oak::crypto {
+
+namespace {
+using oak::crypto::v1::EncryptedRequest;
+using oak::crypto::v1::EncryptedResponse;
+}  // namespace
+
+absl::StatusOr<std::unique_ptr<ClientEncryptor>> ClientEncryptor::Create(
+    absl::string_view serialized_server_public_key) {
+  absl::StatusOr<SenderContext> setup_base_sender_result =
+      SetupBaseSender(serialized_server_public_key, OAK_HPKE_INFO);
+  if (!setup_base_sender_result.ok()) {
+    return setup_base_sender_result.status();
+  }
+  SenderContext sender_context = std::move(*setup_base_sender_result);
+  return std::make_unique<ClientEncryptor>(ClientEncryptor(sender_context));
+}
+
+absl::StatusOr<std::string> ClientEncryptor::Encrypt(absl::string_view plaintext,
+                                                     absl::string_view associated_data) {
+  // Encrypt request.
+  absl::StatusOr<std::string> seal_result =
+      sender_request_context_->Seal(plaintext, associated_data);
+  if (!seal_result.ok()) {
+    return seal_result.status();
+  }
+  std::string ciphertext = std::move(*seal_result);
+
+  // Create request message.
+  EncryptedRequest request;
+  *request.mutable_encrypted_message()->mutable_ciphertext() = ciphertext;
+  *request.mutable_encrypted_message()->mutable_associated_data() = associated_data;
+
+  // Encapsulated public key is only sent in the initial request message of the session.
+  if (!serialized_encapsulated_public_key_.empty()) {
+    *request.mutable_serialized_encapsulated_public_key() = serialized_encapsulated_public_key_;
+    serialized_encapsulated_public_key_ = "";
+  }
```

Comment:
This seems a little sketchy... What if the class has a static boolean variable to determine if it has sent the encap public key already?